server/snapshot/snapshot_test.go

```diff
@@ -0,0 +1,209 @@
+package snapshot
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/docker/notary/server/storage"
+	"github.com/docker/notary/tuf/data"
+	"github.com/docker/notary/tuf/signed"
+)
+
+func TestSnapshotExpired(t *testing.T) {
+	sn := &data.SignedSnapshot{
+		Signatures: nil,
+		Signed: data.Snapshot{
+			Expires: time.Now().AddDate(-1, 0, 0),
+		},
+	}
+	assert.True(t, snapshotExpired(sn), "Snapshot should have expired")
+}
+
+func TestSnapshotNotExpired(t *testing.T) {
+	sn := &data.SignedSnapshot{
+		Signatures: nil,
+		Signed: data.Snapshot{
+			Expires: time.Now().AddDate(1, 0, 0),
+		},
+	}
+	assert.False(t, snapshotExpired(sn), "Snapshot should NOT have expired")
+}
+
+func TestGetSnapshotKeyCreate(t *testing.T) {
+	store := storage.NewMemStorage()
+	crypto := signed.NewEd25519()
+	k, err := GetOrCreateSnapshotKey("gun", store, crypto, data.ED25519Key)
+	assert.Nil(t, err, "Expected nil error")
+	assert.NotNil(t, k, "Key should not be nil")
+
+	k2, err := GetOrCreateSnapshotKey("gun", store, crypto, data.ED25519Key)
+
+	assert.Nil(t, err, "Expected nil error")
+
+	// trying to get the same key again should return the same value
+	assert.Equal(t, k, k2, "Did not receive same key when attempting to recreate.")
+	assert.NotNil(t, k2, "Key should not be nil")
+}
```

Comment:
Ok, that's fair enough, especially since the tests for `GetOrCreateSnapshot` also test signing, which can't be done without the key being in the crypto service.  It might be useful to check that `CryptoService.Create` isn't called twice, and getting the key from the store is just masking the second create, but as that doesn't actually hurt anything if the code did that, it makes sense not to check it.